test/realtime/message.test.js

```diff
@@ -917,6 +917,119 @@ define(['ably', 'shared_helper', 'async', 'chai'], function (Ably, helper, async
+          [
+            {
+              name: 'correct',
+              extras: {
+                ref: {
+                  type: 'com.ably.test',
+                  timeserial: '0123456789',
+                },
+              },
+            },
+            {
+              name: 'incorrect-noref',
+            },
+            {
+              name: 'incorrect-badtype',
+              extras: {
+                ref: {
+                  type: 'com.ably.incorrect',
+                  timeserial: '0123456789',
+                },
+              },
+            },
+            {
+              name: 'incorrect-badid',
+              extras: {
+                ref: {
+                  type: 'com.ably.test',
+                  timeserial: '000000000000',
+                },
+              },
+            },
+          ],
+          cb
+        );
+      }
+
+      function subscribe(cb) {
+        channel.subscribe(
+          {
+            refType: 'com.ably.test',
+            refTimeserial: '0123456789',
+          },
+          (m) => {
+            try {
+              expect(m.name).to.be.equal('correct', 'Correct message received');
+            } catch (e) {
+              return cb(e);
+            }
+            // Wait for any errant messages to arrive before continuing
+            config.nextTick(cb);
+          }
+        );
+      }
+
+      async.series(
+        [
+          (cb) => realtime.connection.once('connected', () => cb()),
+          (cb) => channel.attach(cb),
+          (cb) => async.parallel([subscribe, send], cb),
+        ],
+        (err) => closeAndFinish(done, realtime, err)
+      );
+    });
+
+    it('unsubscribe_with_filter_object', function (done) {
+      const realtime = helper.AblyRealtime();
+      const channel = realtime.channels.get('unsubscribe_with_filter_object');
+
+      function send(cb) {
+        channel.publish(
+          [
+            {
+              name: 'incorrect',
+              extras: {
+                ref: {
+                  type: 'com.ably.test',
+                  timeserial: '0123456789',
+                },
+              },
+            },
+          ],
+          cb
+        );
+      }
+
+      function unsubscribe(cb) {
+        try {
+          const listener = () => expect.fail('Listener should not fire');
+          channel.subscribe({ refType: 'com.ably.test', refTimeserial: '0123456789' }, listener);
+          expect(channel.filteredSubscriptions.has(listener), 'Listener should initially be present').to.be.true;
+          channel.unsubscribe(listener);
+          expect(
+            channel.filteredSubscriptions.has(listener),
+            'Listener should no longer be present after unsubscribing'
+          ).to.be.false;
+          config.nextTick(cb);
+        } catch (e) {
+          cb(e);
+        }
+      }
+
+      async.series(
+        [(cb) => realtime.connection.once('connected', () => cb()), (cb) => channel.attach(cb), unsubscribe, send],
```

Comment:
These test fixtures should run on IE so avoid arrow functions (and any other es6+ syntax). We should probably add an eslint config for the test directory tbh.

Reply:
[Done](https://github.com/ably/ably-js/pull/1003/commits/19a22ec2d1d673c99156dd215699a1de6a307865)